internal/vcs/git/refs.go

```diff
@@ -336,3 +337,15 @@ func showRef(ctx context.Context, repo gitserver.Repo, args ...string) ([]Ref, e
 	}
 	return refs, nil
 }
+
+var invalidBranch = lazyregexp.New(`\.\.|/\.|\.lock$|[\000-\037\177 ~^:?*[]+|^/|/$|//|\.$|@{|^@$|\\`)
+
+// ValidateBranchName returns an error if the given string is not a valid branch name.
+// It follows the rules here: https://git-scm.com/docs/git-check-ref-format
+// NOTE: It does not require a slash as mentioned in point 2.
+func ValidateBranchName(branch string) error {
+	if invalidBranch.MatchString(branch) {
+		return errors.New("invalid git branch")
+	}
+	return nil
+}
```

Comment:
Nitpicky suggestion, feel free to ignore, but I don't think returning an error makes a lot of sense. "invalid git branch" doesn't contain any valuable information that needs to be passed up and right now you're throwing it away.

```suggestion
func ValidBranch(branch string) bool {
	return !invalidBranch.MatchString(branch)
}
```